model/github_app_auth.go

```diff
@@ -0,0 +1,78 @@
+package model
+
+import (
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/mongodb/anser/bsonutil"
+	adb "github.com/mongodb/anser/db"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+var (
+	githubAppAuthIdKey      = bsonutil.MustHaveTag(GithubAppAuth{}, "Id")
+	githubAppAuthAppId      = bsonutil.MustHaveTag(GithubAppAuth{}, "AppId")
+	githubAppAuthPrivateKey = bsonutil.MustHaveTag(GithubAppAuth{}, "PrivateKey")
+)
+
+const (
+	GitHubAppAuthCollection = "github_app_auth"
+)
+
+// GithubAppAuth hold the appId and privateKey for the github app associated with the project.
+// It will not be stored along with the project settings, instead it is fetched only when needed
+type GithubAppAuth struct {
+	//Should match the identifier of the project it refers to
+	Id string `bson:"_id" json:"_id"`
+
+	AppId      int64  `bson:"app_id" json:"app_id"`
+	PrivateKey []byte `bson:"private_key" json:"private_key"`
+}
+
+func FindOneGithubAppAuth(projectId string) (*GithubAppAuth, error) {
+	githubAppAuth := &GithubAppAuth{}
+	q := db.Query(bson.M{githubAppAuthIdKey: projectId})
+	err := db.FindOneQ(GitHubAppAuthCollection, q, githubAppAuth)
+	if adb.ResultsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	return githubAppAuth, nil
+}
+
+func HasGithubAppAuth(projectId string) (bool, error) {
```

Comment:
Could you document the exported function?-

Also what do you think about HasGithubAppAuth just calling FindOneGithubAppAuth under the hood and checking if the returned pointer is nil or not? I think it removes some code duplication but it's up to you if you think we should switch it!